app/ui/node-settings/browser.js

```diff
@@ -0,0 +1,310 @@
+(function () {
+  'use strict'
+  const handlebars = require('handlebars')
+  const uuid4      = require('uuid4')
+  const i18n       = require('../../lib/i18n.js')
+
+  const logger       = require('../../lib/logger.js');
+  const clientConfig = require('../../lib/config.js');
+  const syncFactory  = require('@gyselroth/balloon-node-sync');
+
+  const electron = require('electron');
+  const ipcRenderer = electron.ipcRenderer;
+
+  handlebars.registerHelper('i18n', (key) => {
+    var translation = i18n.__(key);
+
+    return new handlebars.SafeString(translation);
+  })
+
+  var sync,
+      promise = new Promise((resolve) => {
+        if (!clientConfig.get('authMethod')) {
+          logger.info('AUTH: no authentication method set yet');
+          return resolve();
+        }
+
+        clientConfig.retrieveSecret(clientConfig.getSecretType()).then((secret) => {
+          clientConfig.setSecret(secret);
+          resolve();
+        }).catch((error) => {
+          logger.error('AUTH: failed retrieve secret from keystore', {error});
+          resolve();
+        });
+      });
+
+  promise.then(() => {
+    initNodeSettings();
+  });
+
+  function initNodeSettings() {
+    sync = syncFactory(clientConfig.getAll(true), logger);
+
+    $(document).ready(() => {
```

Comment:
document.ready seems weird here and the request (sync.blnApi.getAttributes) can be sent before the dom has been rendered.